src/query/catalog/src/database.rs

```diff
@@ -41,4 +63,115 @@ pub trait Database: DynClone + Sync + Send {
     async fn init_database(&self, _tenant: &str) -> Result<()> {
         Ok(())
     }
+
+    // Build a `Arc<dyn Table>` from `TableInfo`.
+    fn get_table_by_info(&self, _table_info: &TableInfo) -> Result<Arc<dyn Table>> {
+        Err(ErrorCode::UnImplement(format!(
+            "UnImplement get_table_by_info in {} Database",
+            self.name()
+        )))
+    }
+
+    // Get one table by db and table name.
+    async fn get_table(
+        &self,
+        _tenant: &str,
+        _db_name: &str,
+        _table_name: &str,
+    ) -> Result<Arc<dyn Table>> {
```

Comment:
Such an API looks strange: `self` is already a `Database`, but getting a `Table` requires another `db_name: str`.

This API suggests a `Database` implementation does not know what itself is. And for developers who implement a `Database`, it is also confusing: what should `get_table()` do if the arg `db_name` is different from the name of `self`?

Reply:
Agree.